enterprise/internal/codeintel/bundles/persistence/postgres/migrate.go

```diff
@@ -52,9 +53,13 @@ func migrateMeta(ctx context.Context, from *sqlitestore.Store, to dbutil.DB, dum
 		return err
 	}
 
-	return withBatchInserter(ctx, to, "lsif_data_metadata", []string{"dump_id", "num_result_chunks"}, func(inserter *BatchInserter) error {
-		for _, value := range values {
-			if err := inserter.Insert(ctx, dumpID, value); err != nil {
+	ch := make(chan int, 1)
+	ch <- values[0]
+	close(ch)
+
+	return withBatchInserter(ctx, to, "lsif_data_metadata", []string{"dump_id", "num_result_chunks"}, func(inserter *batch.BatchInserter) error {
+		for v := range ch {
+			if err := inserter.Insert(ctx, dumpID, v); err != nil {
 				return err
 			}
 		}
```

Comment:
Curious why not just this?

```suggestion
	return withBatchInserter(ctx, to, "lsif_data_metadata", []string{"dump_id", "num_result_chunks"}, func(inserter *batch.BatchInserter) error {
		if err := inserter.Insert(ctx, dumpID, values[0]); err != nil {
			return err
		}
```

Reply:
`withBatchInserter` parallelizes so if we don't deduplicate work via channel we'll upload dupliclates.

Reply:
what a workaround :eyes: at most maybe comment that so its more obvious to the next person reading :)